### PR TITLE
More JS cleanup

### PR DESCRIPTION
**Pull request by dug-stripe**
Manually fixes violations for three more eslint rules (also removes two more that no longer had any violations).

The major change here is to move all React components to their own files. This leads to a significant improvement in code comprehensibility. (A lot of these could be converted to [stateless functional components](https://reactjs.org/blog/2015/10/07/react-v0.14.html#stateless-functional-components) but one thing at a time).

Tested by deploying to edge.

r? @stripe/data-platform

### CODE REVIEW

**Comment by dug-stripe** [comment]
updated PR to fix no-return-assign violation

**Comment by dug-stripe** [comment]
Updated PR to fix react/no-find-dom-node violations. This is deployed to edge.

File: .eslintrc.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
         "react"
     ],
     "rules": {
-        "class-methods-use-this": "warn",
+        "class-methods-use-this": ["error", {"exceptMethods": ["render"]}],
```
Comment (connie-stripe) [inline .eslintrc.json]: does this exclude a method from this rule? if so, why?

**Comment by connie-stripe** [comment]
lgtm

File: .eslintrc.json (hunk 1 above)
Comment (dug-stripe) [reply .eslintrc.json]: good question! `render` is a react lifecycle method that can't be made static.

**Pull request merged**